### PR TITLE
fix: use proper state to verify attestations

### LINKED ISSUE #4831: Wrong state in attestation validation
**Describe the bug**

`packages/beacon-node/src/chain/validation/attestation.ts:81` retrieves the wrong state.
- Must be a state in the same chain as attHeadBlock, but dialed to target.epoch

It causes signature validation to fail if the first slot of a fork is missed. The domain of the state will be wrong, since it will pick the domain of the previous fork, due to updated getDomain logic

### LINKED ISSUE #5498: Use proper state to verify attestations
- Now that we have https://github.com/ChainSafe/lodestar/pull/5390, we may be able to re-enable https://github.com/ChainSafe/lodestar/pull/4849
- use `maxSkipSlots` as the parameter to avoid regen rather than just using a state at the wrong epoch (expose via CLI too)
- related https://github.com/ChainSafe/lodestar/issues/4831

### PR DESCRIPTION
**Pull request by twoeths**
**Motivation**

Right now we always use head state to verify attestations

**Description**

We should only use head block state if:
- head block slot and attestation slot are in the same fork
- and head block state includes committees of target epoch

otherwise we should use head state dialed to target epoch. Notice that with `maxSkippedSlot=32` we'll always use head state unless it's at epoch boundary

Also add some metrics to monitor slot distances in attestation

Closes #5498 Closes #4831

**TODOs**
- [x] Unit test
- [x] Deploy and monitor on a test mainnet node

### CODE REVIEW

**Comment by twoeths** [comment]
Tested for some days on a mainnet node `lodestar_gossip_attestation_use_head_block_state_dialed_to_target_epoch_count` shows no value. Ready for review.

File: packages/beacon-node/src/chain/validation/aggregateAndProof.ts [hunk 1]
```diff
@@ -106,14 +117,13 @@ export async function validateGossipAggregateAndProof(
   // Using the target checkpoint state here caused unstable memory issue
   // See https://github.com/ChainSafe/lodestar/issues/4896
   // TODO: https://github.com/ChainSafe/lodestar/issues/4900
```
Comment (wemeetagain) [inline packages/beacon-node/src/chain/validation/aggregateAndProof.ts]:
```suggestion
```

File: packages/beacon-node/src/chain/validation/attestation.ts [hunk 2]
```diff
@@ -163,14 +175,13 @@ export async function validateGossipAttestation(
     // Using the target checkpoint state here caused unstable memory issue
     // See https://github.com/ChainSafe/lodestar/issues/4896
     // TODO: https://github.com/ChainSafe/lodestar/issues/4900
```
Comment (wemeetagain) [inline packages/beacon-node/src/chain/validation/attestation.ts]:
```suggestion
```

**Review by wemeetagain — COMMENTED** [review]
remove stale comments

**Review by wemeetagain — APPROVED** [review]
lgtm

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.9.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.9.0) :tada:

**Pull request merged**